player/command.c

```diff
@@ -906,8 +906,8 @@ static int mp_property_chapter(void *ctx, struct m_property *prop,
         } else // Absolute set
             step_all = *(int *)arg - chapter;
         chapter += step_all;
-        if (chapter < -1)
-            chapter = -1;
+        if (chapter < 0) // avoid using -1 if first chapter starts at 0
+            chapter = (chapter_start_time(mpctx, 0) <= 0) ? 0 : -1;
```

Comment:
Is there a reason to not just clamp the value in `get_current_chapter`?

Reply:
I think there might be a better approach. It seems like the only time that `last_chapter_seek` doesn't equal -2 is when the user requests a chapter seek, so we can just say:
```c
return (mpctx->last_chapter_seek > -2) ? mpctx->last_chapter_seek : i - 1;
```

Reply:
Just to comment a bit more on this specific section of code that you've highlighted, currently a scenario like this would just be ignored. This is one of the advantages of using `MPMAX`. It makes sense to have a bias for larger chapter indexes as the natural inclination is for chapter indexes to get bigger over time.

With this new approach, what ends up happening is `last_chapter_seek` gets set to -1, and then it waits for the actual current chapter to equal -1 as well, but unless there's a distinct segment that comes before the first chapter, that'll never happen, the current chapter will never equal -1, and `last_chapter_seek` will get stuck waiting forever or until a new chapter is requested by the user.